packages/libraries/core/src/components/tooltip/tooltip/ods-tooltip-controller.spec.ts

```diff
@@ -0,0 +1,270 @@
+import { OdsClearLoggerSpy, OdsInitializeLoggerSpy, OdsLoggerSpyReferences } from '@ovhcloud/ods-testing/src';
+import { Ods } from '../../../configure/ods';
+import { OdsLogger } from '../../../logger/ods-logger';
+import { HTMLStencilElement } from '@stencil/core/internal';
+import { OdsTooltip } from './ods-tooltip';
+import { OdsTooltipController } from './ods-tooltip-controller';
+import { OdsTooltipVariant } from './ods-tooltip-variant';
+import { OdsTooltipMock } from './ods-tooltip.mock';
+import { OcdkSurface, OcdkSurfaceMock } from '@ovhcloud/ods-cdk';
+
+describe('spec:ods-tooltip-controller', () => {
+  let controller: OdsTooltipController;
+  let component: OdsTooltip;
+  let loggerSpyReferences: OdsLoggerSpyReferences;
+
+  Ods.instance().logging(false);
+
+  function setup(attributes: Partial<OdsTooltip> = {}) {
+    component = { ...component, ...attributes };
+    controller = new OdsTooltipController(component);
+  }
+
+  beforeEach(() => {
+    component = new OdsTooltipMock();
+    component.el = document.createElement('osds-tooltip') as HTMLStencilElement;
+
+    const loggerMocked = new OdsLogger('myLoggerMocked');
+    loggerSpyReferences = OdsInitializeLoggerSpy({
+      loggerMocked: loggerMocked as never,
+      spiedClass: OdsTooltipController,
+    });
+  });
+
+  afterEach(() => {
+    OdsClearLoggerSpy(loggerSpyReferences);
+    jest.clearAllMocks();
+  });
+
+  it('should initialize', () => {
+    setup();
+    expect(controller).toBeTruthy();
+  });
+
+  describe('methods', () => {
+    beforeEach(() => {
+      setup(component);
+    });
+
+    describe('checkForClickOutside', () => {
+      const target = document.createElement('button');
+
+      it('should do nothing if there is no surface', () => {
+        expect(() => {
+          controller.checkForClickOutside({ target });
+        }).not.toThrow();
+        expect(component.surface).toBeUndefined();
+      });
+
+      it('should do nothing if surface is not opened', () => {
+        component.surface = new OcdkSurfaceMock() as unknown as OcdkSurface;
```

Comment:
we can't cast directly in OcdkSurface ?

Reply:
`neither type sufficiently overlaps with the other`
Though it seems to be doable as it's done on popover `https://github.com/ovh/design-system/pull/14/files#diff-c2485612538bedda38b90d6226829f889891f5cc62e1d713da2685976f0c0a56` and the setup is quite similar

Any idea what may be missing @skhamvon @astagnol ?